lib/posix/eventfd.c

```diff
@@ -17,6 +17,8 @@
 #define EFD_FLAGS_SET_INTERNAL (EFD_SEMAPHORE | EFD_NONBLOCK)
 
 struct eventfd {
+	struct k_poll_signal read_sig;
```

Comment:
Hmmm.. I was really hoping to avoid these two. Is there no other `poll()` notification mechanism?

Reply:
AFAIK this is one of the most lightweight mechanisms. At least nothing better comes to my mind right now.